api/gdpr.py

```diff
@@ -0,0 +1,165 @@
+from typing import List
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.shortcuts import get_object_or_404
+from helsinki_gdpr.models import SerializableMixin
+from helsinki_gdpr.views import DeletionNotAllowed, DryRunSerializer, GDPRAPIView
+from rest_framework import status
+from rest_framework.response import Response
+
+from tilavarauspalvelu.utils.anonymisation import anonymize_user_data
+
+User = get_user_model()
+
+
+class ProfileUser(SerializableMixin):
+    def __init__(self, user: User, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.user = user
+
+    serialize_fields = (
+        {"name": "user", "accessor": lambda u: u.get_full_name()},
+        {"name": "email"},
+        {"name": "reservations"},
+        {"name": "applications"},
+    )
+
+    @property
+    def reservations(self) -> List[List]:
+        reservations = []
+        for res in self.user.reservation_set.all():
+            reservations.append(
+                [
+                    res.name,
+                    res.description,
+                    res.reservee_first_name,
+                    res.reservee_last_name,
+                    res.reservee_email,
+                    res.reservee_phone,
+                    res.reservee_address_zip,
+                    res.reservee_address_city,
+                    res.reservee_address_street,
+                    res.billing_first_name,
+                    res.billing_last_name,
+                    res.billing_email,
+                    res.billing_phone,
+                    res.billing_address_zip,
+                    res.billing_address_city,
+                    res.billing_address_street,
+                    res.reservee_id,
+                    res.reservee_organisation_name,
+                    res.free_of_charge_reason,
+                    res.cancel_details,
+                ]
+            )
+        return reservations
+
+    @property
+    def applications(self) -> List:
+        applications = []
+        for app in self.user.application_set.all():
```

Comment:
Black does not enforce spacing, but in a few places here, I believe spacing makes code more readable. I'll comment on each line that I believe padding will help.

Here for example, an empty line before the `for` loop makes it more clear that a new section of code starts.

Reply:
I partly agree with some spacing we could improve readability. Though I would see it too airy to put it like:
```
applications = []

for app in self.user.application_set.all():
        applications.append(app.additional_information)
```
since the `applications = []` is a variable associated with the for loop and thus make sense (to me) to wrap it firmly (:heart: ) with for loop since it makes "logical whole" (again _to me_) 

For those other spacings I'll answer below.

Reply:
Yeah, this is stuff that coders could argue about till the end of times!

I see your point here and will not argue that padding is a must here.

I see `applications` as a "global" variable within the function, since it is also the variable that is returned at the end of the function. The for loop is then the "builder" that assembles the output to the function. That is why I would like it stand-alone at the top of the function followed by padding before the for loop.

But, I see your point. This is getting more philosophical than necessary :smile: